lib/services/cloud-platform-service.ts

```diff
@@ -0,0 +1,120 @@
+import * as semver from "semver";
+import * as constants from "../constants";
+
+export class CloudPlatformService implements ICloudPlatformService {
+	private shouldUseOldPlatformService: boolean;
+
+	constructor(
+		private $injector: IInjector,
+		private $staticConfig: IStaticConfig
+	) {
+		const cliVersion = this.$staticConfig.version;
+		this.shouldUseOldPlatformService = semver.valid(cliVersion) && semver.lt(cliVersion, semver.prerelease(cliVersion) ? "5.4.0-2019-05-16-13277" : "6.0.0");
+	}
+
+	public get $nsCloudBuildCommandHelper(): IBuildCommandHelper {
+		return this.$injector.resolve("nsCloudBuildCommandHelper");
```

Comment:
Why we don't let the injector resolve this dependency?

Reply:
Because there is a cyclic dependency.